pkg/cloud-init/cloud-init.go

```diff
@@ -162,52 +163,20 @@ func ReadCloudInitVolumeDataSource(vmi *v1.VirtualMachineInstance, secretSourceD
 	return nil, nil
 }
 
-// resolveNoCloudSecrets is looking for CloudInitNoCloud volumes with UserDataSecretRef
-// requests. It reads the `userdata` secret the corresponds to the given CloudInitNoCloud
-// volume and sets the UserData field on that volume.
-//
-// Note: when using this function, make sure that your code can access the secret volumes.
-func resolveNoCloudSecrets(vmi *v1.VirtualMachineInstance, secretSourceDir string) error {
-	volume := findCloudInitNoCloudSecretVolume(vmi.Spec.Volumes)
-	if volume == nil {
-		return nil
-	}
-
-	baseDir := filepath.Join(secretSourceDir, volume.Name)
-	var userDataError, networkDataError error
-	var userData, networkData string
-	if volume.CloudInitNoCloud.UserDataSecretRef != nil {
-		userData, userDataError = readFirstFoundFileFromDir(baseDir, []string{"userdata", "userData"})
-	}
-	if volume.CloudInitNoCloud.NetworkDataSecretRef != nil {
-		networkData, networkDataError = readFirstFoundFileFromDir(baseDir, []string{"networkdata", "networkData"})
-	}
-	if userDataError != nil && networkDataError != nil {
-		return fmt.Errorf("no cloud-init data-source found at volume: %s", volume.Name)
-	}
-
-	if userData != "" {
-		volume.CloudInitNoCloud.UserData = userData
-	}
-	if networkData != "" {
-		volume.CloudInitNoCloud.NetworkData = networkData
-	}
+func isNotNoCloudAccessCredential(accessCred v1.AccessCredential) bool {
```

Comment:
Can you change it to be not negated? I.e. to `isNoCloudAccessCredential`?

Reply:
2 against 1, reverted it back 😄 
@lyarwood, fyi

ref https://github.com/kubevirt/kubevirt/pull/10231#discussion_r1303986821